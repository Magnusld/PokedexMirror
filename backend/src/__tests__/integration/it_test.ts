import {createTestContext} from "./__helpersTestEnvironment";
import {gql} from "graphql-request";

jest.setTimeout(30000)

const ctx = createTestContext();

test("mutation createRating returns and persists on server", async () => {
    const result =
    await ctx.client.request(gql`mutation CreateRatingMutation($data: RatingCreateInput!) {
                CreateRating(data: $data) {
                    id
                    pokemonId
                    ratedPokemon { id name }
                    userGuid
                }
            }`,
        {
            data: {
                pokemonId: 1,
                userGuid: "testrating for pokemon1",
                rating: 5,
            }
        }
    );
    expect(result).toMatchSnapshot()

    const persistedData = await ctx.db.pokemonRating.findMany()
    expect(persistedData).toMatchSnapshot()
});

test.todo//("", async () => {})