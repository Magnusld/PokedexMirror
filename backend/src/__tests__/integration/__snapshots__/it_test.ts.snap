// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutation createRating returns and persists on server 1`] = `
Object {
  "CreateRating": Object {
    "id": 1,
    "pokemonId": 1,
    "ratedPokemon": Object {
      "id": 1,
      "name": "TestPokemon2",
    },
    "userGuid": "testrating for pokemon1",
  },
}
`;

exports[`mutation createRating returns and persists on server 2`] = `
Array [
  Object {
    "id": 0,
    "pokemonId": 0,
    "rating": 5,
    "userGuid": "testGuid",
  },
  Object {
    "id": 1,
    "pokemonId": 1,
    "rating": 5,
    "userGuid": "testrating for pokemon1",
  },
]
`;
